torch/jit/__init__.py

```diff
@@ -399,6 +400,32 @@ def script_method(fn):
     return ScriptMethodStub(createResolutionCallback(frames_up=2), get_jit_ast(fn), fn)
 
 
+def batch(batch_size=1, optimize=True, _frames_up=0):
+    def decorator(fn):
+        mod = script(fn, optimize, _frames_up)
+        res_graph = mod.to_batch(batch_size)
```

Comment:
We should do this as:

`res_graph = torch.C._jit_batch(mod.graph, batch_size)`

Otherwise it couples all of the Module infrastructure  to batching, but this stuff has nothing to do with the batching transform.